okhttp/src/main/java/okhttp3/Challenge.java

```diff
@@ -16,67 +16,92 @@
 package okhttp3;
 
 import java.nio.charset.Charset;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Map.Entry;
 import javax.annotation.Nullable;
 
+import static java.util.Collections.singletonMap;
+import static java.util.Collections.unmodifiableMap;
 import static okhttp3.internal.Util.ISO_8859_1;
 
 /** An RFC 7617 challenge. */
```

Comment:
nit: We now refer to this elsewhere as a 7235 challenge.  7617 is Basic auth IIUC

Reply:
You are totally correct.
It was not a pure 7617 challenge before already.
It was any 7235 challenge, but only had their algorithm, realm and charset properties while it now has all their auth params and can be used to implement any auth algorithm.
I changed the comment.